### DISCUSSION TITLE
Best way to prevent parent form re-rendering when the child is updated

### DISCUSSION THREAD

**Discussion by leanjunio** [react-hook-form/react-hook-form#8265] [Q&A] [answered]
I currently have a Parent Form and a Child Form that's sharing the same schema. The Child form is passed the `control` from the parent's `useForm` hook so that it could be used on the child.

The issue that I'm experiencing is that whenever the fields in the child get updated, the parent form re-renders as well.

On the child component, I have a `useWatch` that I'm using to keep watch of the state changes for the fields that are being updated.

Is this happening because they're sharing the same form data object between them?

**Comment by Moshyfawn** [comment, marked as answer]
If you `watch` some state in the parent component and it gets updated wherever _(parent / child)_, both the parent and the child are gonna re-render. It's the way React works.

**Reply by leanjunio** [reply]
Gotcha, I thought there would be a way to keep the whole form from re-rendering.

That being said, the form I'm working with is pretty large and it has a lot of logic, my concern is that re-rendering a large form whenever I change the value of a field from 1.5 to 1.6, the whole form itself would re-render twice.

Would colocating the state to the child component form and only updating the parent state `onSubmit` of the child form be the best workaround for this type of scenario so that the whole parent form doesn't re-render on every keystroke?

**Reply by bluebill1049** [reply]
you can consider to use useWatch and isolate that render.

**Reply by leanjunio** [reply]
Just to clarify, when you say "isolate that render", does that mean that the component that's using that `useWatch` is going to be the only component that will re-render?

**Reply by bluebill1049** [reply]
yes
